DigiMeSDK/Core/Classes/Entities/CAFile.swift

```diff
@@ -0,0 +1,179 @@
+    }
+}
+
+internal class CAImageData: CADataRepresentation {
+    
+    public typealias FileDataType = UIImage
+    public var fileMimeType: CAMimeType
+    public var fileContent: FileDataType
+    
+    public static let compatibleMimeTypes: [CAMimeType] = [.image_bmp, .image_gif, .image_png, .image_jpeg, .image_tiff]
+    
+    public required init(rawData: Data, mimeType: CAMimeType) throws {
+        
+        guard let image = UIImage(data: rawData) else {
+            throw CADataParsingError.couldNotDeserialiseRawData
+        }
+        
+        self.fileContent = image
+        self.fileMimeType = mimeType
+    }
+}
+
+// CARawData doubles as the ObjC compatibility layer, and hence exposes
+// a number of 'try and do something' methods.
+
+@objcMembers
+internal class CARawData: NSObject, CADataRepresentation {
+    
+    public typealias FileDataType = Data
+    public var fileMimeType: CAMimeType
+    public var fileContent: FileDataType
+    
+    public static let compatibleMimeTypes: [CAMimeType] = CAMimeType.allCases
+    
+    public required init(rawData: Data, mimeType: CAMimeType) throws {
+        self.fileContent = rawData
+        self.fileMimeType = mimeType
+    }
+}
+
+// @available(swift, obsoleted: 0.1, message: "Usage of CAFile is limited to Objective-C. For Swift, which supports generics, use CAFileContainer.")
+@objcMembers
+public class CAFile: NSObject {
+    
+    private var boxedFile: CARawData
+    public var fileId: String
+    public var fileMetadata: CAFileMetadata?
+    
+    public var fileMimeType: CAMimeType {
+        return fileMetadata?.mimeType ?? .application_octetStream
+    }
+    public var fileContent: Data {
+        return boxedFile.fileContent
+    }
+    
+    @available(*, deprecated, message: "Please use -[CAFile fileContentAsJSON] instead.")
+    @available(*, renamed: "fileContentAsJSON")
+    public var json: [[AnyHashable: Any]]? {
+        return fileContentAsJSON()
+    }
+    
+    public override var description: String {
+        let objCount = fileContentAsJSON()?.count ?? -1
+        return "File ID: \(fileId), \(objCount >= 0 ? "\(objCount) objects." : "\(fileContent.count) bytes.")"
+    }
+    
+    public init(fileId: String, fileContent: Data, fileMetadata: CAFileMetadata?) {
+        self.fileId = fileId
+        self.boxedFile = try! CARawData(rawData: fileContent, mimeType: fileMetadata?.mimeType ?? .application_octetStream)
+        self.fileMetadata = fileMetadata
+    }
+    
+    public func fileContentAsJSON() -> [[AnyHashable: Any]]? {
+        return (try? JSONSerialization.jsonObject(with: fileContent, options: [])) as? [[AnyHashable: Any]]
+    }
+    
+    public func fileContentAsImage() -> UIImage? {
+        return UIImage(data: fileContent)
+    }
+    
+    public func fileContentAsString() -> String? {
+        return String(data: fileContent, encoding: .utf8)
+    }
+}
+
+internal class CAFileContainer<DataType: CADataRepresentation> {
```

Comment:
Either:
- class is meant to be `internal` in which case it shouldn't have `public` vars/funcs
- class is meant to be `public`